<?xml version="1.0"?>
<robot name="application" xmlns:xacro="http://ros.org/wiki/xacro">

  <xacro:arg name="kinematics_config" default="$(find ur_e_description)/config/ur3e_default.yaml"/>

  <material name="visual_app_mesh">
    <color rgba="${128/255} ${128/255} ${128/255} 1.0"/>
  </material>

  <!-- Common -->
  <link name="world" />
  <link name="app" />
  <joint name="world_app_joint" type="fixed">
    <parent link="world"/>
    <child link="app"/>
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
  </joint>

  <!-- Robot -->
  <xacro:include filename="$(find evd_ros_tasks)/description/ur3e.urdf.xacro"/>
  <joint name="app_robot_common_joint" type="fixed">
    <parent link="app"/>
    <child link="robot_common"/>
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
  </joint>

  <!-- Table -->
  <link name="table_link" type="fixed">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://evd_ros_tasks/description/3d_printer_machine_tending/visual_meshes/Table.stl"/>
      </geometry>
      <material name="visual_app_mesh"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://evd_ros_tasks/description/3d_printer_machine_tending/collision_meshes/Table.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="app_table_joint" type="fixed">
    <parent link="app" />
    <child link="table_link" />
    <origin xyz="0.0 0.36 -0.37" rpy="0.0 0.0 0.0" />
  </joint>

  <!-- 3D Printer -->
  <link name="3d_printer_link" type="fixed">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://evd_ros_tasks/description/3d_printer_machine_tending/visual_meshes/MK2-Printer.stl"/>
      </geometry>
      <material name="visual_app_mesh"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://evd_ros_tasks/description/3d_printer_machine_tending/collision_meshes/MK2-Printer.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="app_3d_printer_joint" type="fixed">
    <parent link="app" />
    <child link="3d_printer_link" />
    <origin xyz="-0.28 0.32 0.3" rpy="0.0 0.0 0.0" />
  </joint>

  <!-- Product Box -->
  <link name="box_link" type="fixed">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://evd_ros_tasks/description/3d_printer_machine_tending/visual_meshes/Box.stl"/>
      </geometry>
      <material name="visual_app_mesh"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://evd_ros_tasks/description/3d_printer_machine_tending/collision_meshes/Box.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="app_box_joint" type="fixed">
    <parent link="app" />
    <child link="box_link" />
    <origin xyz="0.35 0.35 0.07" rpy="0.0 0.0 0.0" />
  </joint>

  <!-- Robot Pedestal -->
  <link name="ur3e_pedestal_link" type="fixed" >
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://evd_ros_tasks/description/3d_printer_machine_tending/visual_meshes/Pedestal.stl"/>
      </geometry>
      <material name="visual_app_mesh"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://evd_ros_tasks/description/3d_printer_machine_tending/collision_meshes/Pedestal.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="robot_common_ur3e_pedestal_joint" type="fixed">
    <parent link="robot_common" />
    <child link="ur3e_pedestal_link" />
    <origin xyz="0.0 0.0 -0.38" rpy="0.0 0.0 0.0" />
  </joint>

</robot>
